From d9679ba56a94b800b91c124eaa88eabb474a407e Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 14 Jul 2022 13:19:01 +0300
Subject: [PATCH 031/102] pcie: Remove the set of pipeP_pclk from S32CC SerDes
 driver

This setting isn't needed as it was part of the initial
SerDes support.

Issue: ALB-8357
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/include/asm/arch-s32/s32-cc/serdes_regs.h | 2 --
 drivers/pci/serdes_s32gen1.c                       | 5 -----
 2 files changed, 7 deletions(-)

diff --git a/arch/arm/include/asm/arch-s32/s32-cc/serdes_regs.h b/arch/arm/include/asm/arch-s32/s32-cc/serdes_regs.h
index 15ffd270ff..d87f52f329 100644
--- a/arch/arm/include/asm/arch-s32/s32-cc/serdes_regs.h
+++ b/arch/arm/include/asm/arch-s32/s32-cc/serdes_regs.h
@@ -142,8 +142,6 @@
 #define PHY_GEN_CTRL_RX_SRIS_MODE_BIT	(9)
 #define PHY_GEN_CTRL_RX_SRIS_MODE_MASK	BIT(PHY_GEN_CTRL_RX_SRIS_MODE_BIT)
 
-#define EXT_PCLK_REQ				(1U << 0)
-
 /* Field definitions for PHY_EXT_MPLLA/B Registers */
 
 #define MPLLA_DIV8_CLK_EN			(1U << 8)
diff --git a/drivers/pci/serdes_s32gen1.c b/drivers/pci/serdes_s32gen1.c
index 814887292e..ea4eaa81aa 100644
--- a/drivers/pci/serdes_s32gen1.c
+++ b/drivers/pci/serdes_s32gen1.c
@@ -238,11 +238,6 @@ static bool s32_serdes_init(struct s32_serdes *pcie)
 	if (ret)
 		return false;
 
-	if (pcie->ss_mode == SERDES_MODE_SGMII_SGMII) {
-		/*	Set pipeP_pclk */
-		W32(UPTR(pcie->dbi) + SS_PHY_GEN_CTRL, EXT_PCLK_REQ);
-	}
-
 	if (s32_serdes_set_mode(pcie->dbi, pcie->ss_mode))
 		return false;
 
-- 
2.17.1

